src/state/client-server-hooks/use-station-filtering-defaults.js

```diff
@@ -0,0 +1,35 @@
+import { useEffect } from "react";
+import { useShallow } from "zustand/react/shallow";
+import { useVariables } from "@/state/query-hooks/use-variables";
+import { useFrequencies } from "@/state/query-hooks/use-frequencies";
+import { useNetworks } from "@/state/query-hooks/use-networks";
+import { useStationsStore } from "@/state/client/stations-store";
+import map from "lodash/fp/map";
+import identity from "lodash/fp/identity";
+
+/**
+ * Integration between query hooks loaded on the Body route and setting of defaults for filters.
+ * @returns
+ */
+export const useStationFilteringDefaults = () => {
+  const { data: variables } = useVariables();
+  const { data: frequencies } = useFrequencies();
+  const { data: networks } = useNetworks();
+  const { setDefaultNetworks, setDefaultVariables, setDefaultFrequencies } =
+    useStationsStore(
+      useShallow((state) => ({
+        setDefaultNetworks: state.setDefaultNetworks,
+        setDefaultVariables: state.setDefaultVariables,
+        setDefaultFrequencies: state.setDefaultFrequencies,
+      })),
+    );
+
+  useEffect(() => {
+    // default setters are guarded against multiple sets, so this is safe to apply on each
+    // network request completion
+    if (variables)
+      setDefaultVariables(map((variable) => variable.id)(variables));
```

Comment:
I'm reluctant to have non-block statements as targets of `if` and the like.

Reply:
Agreed, I think they were one liners but prettier obviously felt I was wrong. I'll block them.

Reply:
React documentation I think indicates the contrary: ["`useEffect` is a React Hook that lets you [synchronize a component with an external system.](https://react.dev/learn/synchronizing-with-effects)"](https://react.dev/reference/react/useEffect). It connects to the React component lifecycle, rather than just avoiding a computation.

React doc has a lot of advice about avoiding using `useEffect` improperly, but this case is a somewhat complicated one. Let's leave it for now and return to it later when there is time to read and discuss at leisure.